### PR TITLE
Add testing for Windows (both local and CI)

### PR DESCRIPTION
**Pull request by dafyddj**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [x] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [x] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Describe the changes you're proposing

Add local testing for Windows using Vagrant.
Add CI testing for Windows using GitHub Actions.
Fix a regression in Windows default values.

### Documentation checklist

- [x] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [x] Included in Kitchen (i.e. under `state_top`).
- [x] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [x] Updated the relevant test pillar.

### CODE REVIEW

**Comment by myii** [comment]
@dafyddj The `Gemfile.lock` adjustments would need to be added to this PR as well, so that the existing Linux testing can actually run.  Will propose some changes to the tests as well, to get them passing the Rubocop violations as well as make them a bit more Rubyish.

File: test/integration/v3000-py3/controls/pkgs_spec.rb [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+if os[:family] == 'windows'
+  pkgs = %w[
+           Salt\ Minion
+         ]
+else
+  pkgs = %w[
+           salt-master
+           salt-minion
+         ]
+end
```
Comment (myii) [inline test/integration/v3000-py3/controls/pkgs_spec.rb:12]:
```suggestion
pkgs =
  case platform[:family]
  when 'windows'
    %w[Salt\ Minion]
  else
    %w[salt-master salt-minion]
  end
```

File: test/integration/v3000-py3/controls/service_spec.rb [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+if os[:family] == 'windows'
+  services = %w[
+               salt-minion
+             ]
+else
+  services = %w[
+               salt-master
+               salt-minion
+             ]
+end
```
Comment (myii) [inline test/integration/v3000-py3/controls/service_spec.rb:12]:
```suggestion
services =
  case platform[:family]
  when 'windows'
    %w[salt-minion]
  else
    %w[salt-master salt-minion]
  end
```

**Comment by myii** [comment]
@dafyddj Note that the `Gemfile.lock` has been updated as part of the weekly maintenance.  Apologies for the extra work required here because of that.

**Comment by myii** [comment]
Checked using https://github.com/myii/ssf-formula/pull/234.

---

Lovely work, @dafyddj -- merged.

One thing from the review above:

> ... To be honest, I'd prefer it even more if these sections could be embedded in the controls themselves, like it's been done here:
> 
> https://github.com/saltstack-formulas/tomcat-formula/blob/4602740178af04242d96583f453d7c50f3dc74e3/test/integration/default/controls/services_spec.rb#L10-L27

* How do you feel about this adjustment being made here?

**Comment by dafyddj** [comment]
Thanks for the merge.

> * How do you feel about this adjustment being made here?

I believe there's a better place for this kind of "configuration" and we should probably look into using [Inputs](https://www.inspec.io/docs/reference/inputs/) instead.

**Pull request merged**